packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts

```diff
@@ -323,8 +327,58 @@ export default createRule({
               }
             }
           }
-        }
 
+          typePartFlags.length >= 2
+            ? seenUnionTypes.set(typeNode, typePartFlags)
+            : null;
+        }
+        const checkIfUnionsAreAssignable = (): undefined => {
+          let typeFlagsOfUnions: ts.TypeFlags[] = [];
+          let result = true;
+          let primitiveUnit: number;
+
+          seenUnionTypes.forEach((value, key) => {
+            value.forEach(union => {
+              typeFlagsOfUnions.push(union.typeFlags);
+            });
```

Comment:
[Style] Is there a need to have a shared variable outside the `forEach`? Switching it to being inside the `.forEach` and never modified doesn't fail any unit tests:

```suggestion
            const typeFlagsOfUnions = unionTypeFlags.flatMap(
              union =>
                union.typeFlags as keyof typeof literalToPrimitiveTypeFlags,
            );
```

That includes an extra tip: since we know the type flags are all the `keyof typeof literalToPrimitiveTypeFlags`, using the assertion early on means we don't need multiple assertions later.

Reply:
I removed the variable, I dont think we need it anymore.  
I might need some help with the type assertions though :sweat_smile: